service/history/replication/ack_manager.go

```diff
@@ -493,6 +493,35 @@ func (p *ackMgrImpl) generateHistoryReplicationTask(
 	)
 }
 
+func (p *ackMgrImpl) generateSyncWorkflowStateTask(
+	ctx context.Context,
+	taskInfo *tasks.SyncWorkflowStateTask,
+) (*replicationspb.ReplicationTask, error) {
+	namespaceID := namespace.ID(taskInfo.NamespaceID)
+	workflowID := taskInfo.WorkflowID
+	runID := taskInfo.RunID
+	taskID := taskInfo.TaskID
+	return p.processReplication(
```

Comment:
not relevant to this PR, but I think we also need to handle namespace not found error inside `processReplication()`. Probably also in replication executor on receiver end.